src/runtime/type.ts

```diff
@@ -1284,15 +1284,14 @@ export class TypeVariableInfo {
   }
 
   isAtLeastAsSpecificAs(other: TypeVariableInfo): boolean {
-    // TODO(mmandlis): add tests for this method!!!
     const thisCanReadSubset = this._canReadSubset || this._originalCanReadSubset;
     const thisCanWriteSuperset = this._canWriteSuperset || this._originalCanWriteSuperset;
     const otherCanReadSubset = other._canReadSubset || other._originalCanReadSubset;
     const otherCanWriteSuperset = other._canWriteSuperset || other._originalCanWriteSuperset;
     return ((!otherCanWriteSuperset && !thisCanWriteSuperset) ||
-            (otherCanWriteSuperset && (!thisCanWriteSuperset || otherCanWriteSuperset.isAtLeastAsSpecificAs(thisCanWriteSuperset)))) &&
+            !!(otherCanWriteSuperset && (!thisCanWriteSuperset || otherCanWriteSuperset.isAtLeastAsSpecificAs(thisCanWriteSuperset)))) &&
```

Comment:
I thought that !! was used to coerce values to booleans - this should already be a boolean and won't require coersion.

Reply:
actually, it's probably just that you need it on otherCanWriteSuperset directly; and likewise for thisCanWriteSuperset.

I think in light of confusion here and immediately below that this should probably be broken up a little for clarity; perhaps even split out into a separate function?

Reply:
as discussed in chat, this method shouldn't be implemented for ranges, removed.